rwalocation/main.py

```diff
@@ -1,39 +1,101 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
-__version__ = "0.1.1"
+__version__ = "0.1.2"
 
-import requests
+import json
 import operator
+from os import path
 
 class Province:
     def __init__(self):
-        self.districts = dict()
-        self.districtsArray = []
-        self.data_url = 'https://raw.githubusercontent.com/DevRW/rwanda-location/develop/src/db/locations.json'
+        self.data_url = path.realpath(path.dirname(path.abspath(__file__)) + '/' + 'data.json')
         self.sorted_data = self.sort_by_district(self.data_url)
+        self.districts = self.district() # the list of all districts equivalent to self.district()
+        self.provinces = self.province() # the list of all provinces equivalent to self.province()
 
     def main(self):
-        self.province()
-        self.district()
-    
+        self.location()
+        self.district('kicukiro')
+        self.location('kag')
+        self.province('East')
+        self.district('kicuk')
+        self.location('kicukiro')
+        self.province('eat')
+
+    def province(self, province_name=None):
+        if province_name:
+            if self.is_province_found(self.provinces, str(province_name)):
+                return self.filter_array(self.sorted_data, 'province_name', str(province_name))
+            return []
+        temp = []
+        provinces_array = []
+        for value in self.sorted_data:
+            if(value['province_name'] not in temp):
+                temp.append(value['province_name'])
+                province_name = value['province_name']
+                province_code = value['province_code']
+                provinces_array.append({'province_name': province_name,'province_code': province_code})
+        return provinces_array
```

Comment:
Hey, Good implementation here...

However, I suggest that we reuse this code in the district methods and other methods that may come up in the future.
Let's have a separate static method that takes in the iterable param (district_name/province_name) and returns an array.
This will improve our code readability and makes it DRYer.

Let me know if it's clear